src/main/java/org/owasp/esapi/Validator.java

```diff
@@ -59,280 +59,438 @@ public interface Validator {
+     * Returns true if {@code input} is valid.
+     * <p>
+     * Calls {@link #getValidSafeHTML(String, String, int, boolean)},
+     * and returns true if no exceptions are thrown.
+     *
+     * @throws IntrusionException Input is clearly an attack.
      */
     boolean isValidSafeHTML(String context, String input, int maxLength, boolean allowNull) throws IntrusionException;
 
     /**
-     * Calls getValidSafeHTML and returns true if no exceptions are thrown.
+     * Returns true if {@code input} is valid,
+     * any validation exceptions are added to the supplied {@code errorList}.
+     * <p>
+     * Calls {@link #getValidSafeHTML(String, String, int, boolean)}
+     * and returns true if no exceptions are thrown.
+     *
+     * @throws IntrusionException Input is clearly an attack.
      */
     boolean isValidSafeHTML(String context, String input, int maxLength, boolean allowNull, ValidationErrorList errorList) throws IntrusionException;
 
     /**
      * Returns canonicalized and validated "safe" HTML that does not contain unwanted scripts in the body, attributes, CSS, URLs, or anywhere else.
+     * <p>
      * Implementors should reference the OWASP AntiSamy project for ideas
      * on how to do HTML validation in a whitelist way, as this is an extremely difficult problem.
      *
      * @param context
-     *         A descriptive name of the parameter that you are validating (e.g., LoginPage_UsernameField). This value is used by any logging or error handling that is done with respect to the value passed in.
+     *         A descriptive name of the parameter that you are validating (e.g., LoginPage_UsernameField).
+     *         This value is used by any logging or error handling that is done with respect to the value passed in.
      * @param input
      *         The actual user input data to validate.
      * @param maxLength
      *         The maximum String length allowed.
      * @param allowNull
-     *         If allowNull is true then an input that is NULL or an empty string will be legal. If allowNull is false then NULL or an empty String will throw a ValidationException.
+     *         If allowNull is true then an input that is NULL or an empty string will be legal.
+     *         If allowNull is false then NULL or an empty String will throw a ValidationException.
      *
      * @return Valid safe HTML
      *
-     * @throws ValidationException
-     * @throws IntrusionException
+     * @throws ValidationException Input is invalid.
+     * @throws IntrusionException Input is clearly an attack.
      */
     String getValidSafeHTML(String context, String input, int maxLength, boolean allowNull) throws ValidationException, IntrusionException;
 
     /**
-     * Calls getValidSafeHTML with the supplied errorList to capture ValidationExceptions
+     * Returns canonicalized and validated "safe" HTML that does not contain unwanted scripts in the body, attributes, CSS, URLs, or anywhere else,
```

Comment:
So I don't want to hold up the merge but I would reference it just so that the programmer is aware of configuration files that would impact execution.

Reply:
Both getValidSafeHtml methods now share the same antisamy words which hopefully emphasises the use of the configuration file and the original emphasis on it not being a trivial problem.